src/main/java/application/MenuController.java

```diff
@@ -122,7 +119,7 @@ public EntityDetailListResponse getDetails(@RequestBody SessionNavigationBean se
             if (detail == null) {
                 throw new RuntimeException("Could not get inline details");
             }
-            return detail;
+            return LocationRelevantResponseBean.setLocationNeeds(detail, menuSession);
```

Comment:
This `LocationRelevantResponseBean` pattern seems un-DRY to me and obscure the meaning of the code (IE this looks like we're returning some location response rather than a case list response that possibly has location details). All of the responses being set here makes calls to `MenuSessionRunnerService` which in turn all call `getDetails` - could this code be centralized there?

Reply:
@wpride I didn't follow the alternative you proposed, but in the most recent commit I moved where the `setLocationNeeds()` helper lives. I think that should resolve the issue you're talking about where it looks like we're returning a different type of response bean. If you want to discuss further what you were suggesting, let me know.